File: mcp-code-intelligence-kotlin/src/main/kotlin/com/codeintel/memory/ingest/autolink/AutoLinkConfig.kt
/** Configuration for auto-linking strategies. KSA-190. */
package com.codeintel.memory.ingest.autolink

import kotlinx.serialization.Serializable

@Serializable
data class SemanticConfig(
    val enabled: Boolean = true,
    val minScore: Double = 0.75,
    val maxEdges: Int = 5
)

@Serializable
data class EntityConfig(
    val enabled: Boolean = true,
    val minJaccard: Double = 0.3,
    val maxEdges: Int = 5
)

@Serializable
data class TagConfig(
    val enabled: Boolean = true,
    val minOverlap: Int = 2,
    val maxEdges: Int = 3
)

@Serializable
data class FtsConfig(
    val enabled: Boolean = true,
    val maxEdges: Int = 3,
    val fallbackThreshold: Int = 2
)

@Serializable
data class AutoLinkConfig(
    val enabled: Boolean = true,
    val semantic: SemanticConfig = SemanticConfig(),
    val entity: EntityConfig = EntityConfig(),
    val tag: TagConfig = TagConfig(),
    val fts: FtsConfig = FtsConfig(),
    val totalMaxEdges: Int = 10
)

/** Default configuration factory. */
fun defaultAutoLinkConfig(): AutoLinkConfig = AutoLinkConfig()

File: backend/mcp-code-intelligence-nodejs/src/memory/auto-link-config.ts
/**
 * AutoLinkConfig — configuration interface and defaults for auto-linking strategies.
 * KSA-190: Auto-Linking Logic on KB Ingest.
 */

export interface AutoLinkConfig {
  enabled: boolean;
  semantic: { enabled: boolean; minScore: number; maxEdges: number };
  entity: { enabled: boolean; minJaccard: number; maxEdges: number };
  tag: { enabled: boolean; minOverlap: number; maxEdges: number };
  fts: { enabled: boolean; maxEdges: number; fallbackThreshold: number };
  totalMaxEdges: number;
}

export function defaultAutoLinkConfig(): AutoLinkConfig {
  return {
    enabled: true,
    semantic: { enabled: true, minScore: 0.75, maxEdges: 5 },
    entity: { enabled: true, minJaccard: 0.3, maxEdges: 5 },
    tag: { enabled: true, minOverlap: 2, maxEdges: 3 },
    fts: { enabled: true, maxEdges: 3, fallbackThreshold: 2 },
    totalMaxEdges: 10,
  };
}
